homeassistant/components/mqtt/client.py

```diff
@@ -498,7 +498,7 @@ async def async_init_client(self) -> None:
     def _async_reader_callback(self, client: mqtt.Client) -> None:
         """Handle reading data from the socket."""
         if (status := client.loop_read(MAX_PACKETS_TO_READ)) != 0:
-            self._async_on_disconnect(status)
+            self._async_handle_callback_exception(status)
```

Comment:
Like https://github.com/home-assistant/core/pull/137613/files#r1945608852

It does not seem to be needed to explicity call the on_disconnect handler.